src/configuration-requestor.ts

```diff
@@ -27,36 +35,70 @@ export default class ConfigurationRequestor {
       createdAt: configResponse.createdAt,
     });
 
-    const flagsHaveBandits = Object.keys(configResponse.bandits ?? {}).length > 0;
+    const flagsHaveBandits = Object.keys(configResponse.banditReferences ?? {}).length > 0;
     const banditStoresProvided = Boolean(
       this.banditVariationConfigurationStore && this.banditModelConfigurationStore,
     );
     if (flagsHaveBandits && banditStoresProvided) {
       // Map bandit flag associations by flag key for quick lookup (instead of bandit key as provided by the UFC)
-      const banditVariations = this.indexBanditVariationsByFlagKey(configResponse.bandits);
+      const banditVariations = this.indexBanditVariationsByFlagKey(configResponse.banditReferences);
 
       await this.hydrateConfigurationStore(this.banditVariationConfigurationStore, {
         entries: banditVariations,
         environment: configResponse.environment,
         createdAt: configResponse.createdAt,
       });
 
-      // TODO: different polling intervals for bandit parameters
-      const banditResponse = await this.httpClient.getBanditParameters();
-      if (banditResponse?.bandits) {
-        if (!this.banditModelConfigurationStore) {
-          throw new Error('Bandit parameters fetched but no bandit configuration store provided');
-        }
+      if (
+        this.requiresBanditModelConfigurationStoreUpdate(
+          this.banditModelVersions,
+          configResponse.banditReferences,
+        )
+      ) {
+        const banditResponse = await this.httpClient.getBanditParameters();
+        if (banditResponse?.bandits) {
+          await this.hydrateConfigurationStore(this.banditModelConfigurationStore, {
+            entries: banditResponse.bandits,
+            environment: configResponse.environment,
+            createdAt: configResponse.createdAt,
+          });
 
-        await this.hydrateConfigurationStore(this.banditModelConfigurationStore, {
-          entries: banditResponse.bandits,
-          environment: configResponse.environment,
-          createdAt: configResponse.createdAt,
-        });
+          this.setBanditModelVersions(
+            this.getLoadedBanditModelVersionsFromStore(this.banditModelConfigurationStore),
+          );
+        }
       }
     }
   }
 
+  private getLoadedBanditModelVersionsFromStore(
+    banditModelConfigurationStore: IConfigurationStore<BanditParameters> | null,
+  ): string[] {
+    if (banditModelConfigurationStore === null) {
+      return [];
+    }
+    return Object.values(banditModelConfigurationStore.entries()).map(
+      (banditParam: BanditParameters) => banditParam.modelVersion,
+    );
+  }
+
+  private setBanditModelVersions(modelVersions: string[]) {
```

Comment:
consider skipping the setter and just using
`this.banditModelVersions = this.getLoadedBanditModels(...)`